src/main/kotlin/org/opensearch/replication/task/shard/ShardReplicationTask.kt

```diff
@@ -193,10 +195,9 @@ class ShardReplicationTask(id: Long, type: String, action: String, description:
         updateTaskState(FollowingState)
         val followerIndexService = indicesService.indexServiceSafe(followerShardId.index)
         val indexShard = followerIndexService.getShard(followerShardId.id)
-        // Adding retention lease at local checkpoint of a node. This makes sure
-        // new tasks spawned after node changes/shard movements are handled properly
-        logInfo("Adding retentionlease at follower sequence number: ${indexShard.lastSyncedGlobalCheckpoint}")
-        retentionLeaseHelper.addRetentionLease(leaderShardId, indexShard.lastSyncedGlobalCheckpoint , followerShardId)
+        // Adding retention lease with retain all. After the task starts getting changes, retention lease would be
+        // renewed based on lastSyncedGlobalCheckpoint.
+        retentionLeaseHelper.addRetentionLease(leaderShardId, RetentionLeaseActions.RETAIN_ALL , followerShardId)
```

Comment:
should we do `renewRetentionLease` here ?

Reply:
+1

Reply:
Done